Sources/Fullscreen/TransactionView/TransactionStepView/TransactionStepView.swift

```diff
@@ -212,25 +215,67 @@ private extension TransactionStepView {
         if let buttonModel = buttonModel {
             let buttonText = buttonModel.text
             let buttonStyle = TransactionStepView.ActionButton(rawValue: buttonModel.style).style
+            let buttonAction = TransactionStepView.ActionButton.Action(rawValue: buttonModel.action ?? "")
 
             let button = Button(style: buttonStyle, withAutoLayout: true)
             button.setTitle(buttonText, for: .normal)
             button.isEnabled = style.actionButtonEnabled
+            button.titleLabel?.numberOfLines = 0
             button.tag = tag
             button.addTarget(self, action: #selector(handleButtonTap(_:)), for: .touchUpInside)
             button.setContentHuggingPriority(.required, for: .vertical)
 
-            if model.state == .completed {
-                button.contentHorizontalAlignment = .leading
-                button.contentEdgeInsets = .leadingInset(4)
-            }
-
             verticalStackView.addArrangedSubview(button)
             verticalStackView.setCustomSpacing(.spacingM, after: button)
 
+            switch buttonAction {
+            case .seeAd:
+                button.contentHorizontalAlignment = .leading
+                button.contentEdgeInsets = .leadingInset(.spacingS)
+            default:
+                addWebViewIconToButton(button)
+            }
+
+            button.leadingAnchor.constraint(equalTo: titleView.leadingAnchor).isActive = true
             bottomAnchorConstraint = bottomAnchor.constraint(equalTo: button.bottomAnchor, constant: .spacingM)
         }
     }
+
+    func addWebViewIconToButton(_ button: Button) {
+        let imageView = UIImageView(image: UIImage(named: .webview))
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        let imageWidth: CGFloat = 16
+        var leadingInset = CGFloat.zero
+
+        if model.state == .completed {
+            button.contentHorizontalAlignment = .leading
+            leadingInset = -10
```

Comment:
Why this inset if its state is `.completed`? 😅

Reply:
If the state of the step is `.notStarted` or `.completed` all the labels and button labels should be vertically aligned with the `leadingAnchor` of the title. If the state of the step is `.active` then everything should be indented inwards toward the centre.

This button has the additional icon along with custom leading and trailing insets to centre in correctly. If I don't provide it with a negative inset the `titleLabel` in the button looks a few points off compared to the first step 😄

Reply:
Ok, thanks for the explanation :) My cabin fevered mind couldn't figure it out why this was needed 😅